test/unit/ui.js

```diff
@@ -212,5 +182,143 @@ describe('ui class', () => {
         assert.calledOnce(testCart.destroy);
       });
     });
+
+    describe('cart methods', () => {
+      describe('createCart', () => {
+        let initStub;
+
+        beforeEach(() => {
+          initStub = sinon.stub(Cart.prototype, 'init').resolves();
+        });
+
+        afterEach(() => {
+          initStub.restore();
+        });
+
+        describe('when no cart exists', () => {
+          it('creates a new cart', async () => {
+            await ui.createCart({options: {}});
+            assert.equal(1, ui.components.cart.length);
+            assert.instanceOf(ui.components.cart[0], Cart);
+            assert.calledOnce(initStub);
+            ui.destroyComponent('cart', ui.components.cart[0].model.id);
```

Comment:
Is there a way we can remove the dependency on `destroyComponent` by stubbing something out?